nuxeo-retention-core/src/main/java/org/nuxeo/ecm/retention/adapter/Record.java

```diff
@@ -93,4 +114,114 @@ public Calendar getMaxRetentionAt() {
     public void setMaxRetentionAt(Calendar maxRetentionDate) {
         doc.setPropertyValue(RECORD_MAX_RETENTION_AT, maxRetentionDate);
     }
+
+    @SuppressWarnings("unchecked")
+    public void addRule(String ruleId) {
+        List<Map<String, Serializable>> rr = (List<Map<String, Serializable>>) doc.getPropertyValue(RETENTION_RULES);
```

Comment:
What don't you use a getter (exposed from the Adapter) instead of getting the property value